cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -1014,10 +1014,21 @@ func (csr *ClusterStateRegistry) GetUpcomingNodes() (upcomingCounts map[string]i
 	return upcomingCounts, registeredNodeNames
 }
 
+// getRunningNodeGroups returns running node groups, filteres out upcoming ones.
+func (csr *ClusterStateRegistry) getRunningNodeGroups() []cloudprovider.NodeGroup {
+	var result []cloudprovider.NodeGroup
```

Comment:
nit: you know the likely size of `result` here, so might as well allocate enough capacity for it up front. Right now you're going to have O(logN) allocations instead of one, because of calling `append` in a loop like this.